packages/abitype/src/utils.test.ts

```diff
@@ -0,0 +1,20 @@
+import { attest } from '@arktype/attest'
+import { test } from 'vitest'
+import type { TypedDataToPrimitiveTypes } from './utils.js'
+
+test('self-referencing', () => {
```

Comment:
This syntax closely parallels the type assertions vitest allows, with the benefit of being able to run them conditionally based on your environment since the assertion is still executed as normal at runtime.

Another significant benefit is that you can write these assertions alongside or in combination with your value assertions, so you don't need to have separate `.d.ts` tests.

This can be especially helpful when testing types and values that parallel one another, e.g. via APIs like `.throwsAndHasTypeError`

It also allows type/value snapshotting, object literal snapshots etc. (see the admittedly underdocumented [README](https://github.com/arktypeio/arktype/tree/beta/ark/attest#readme)).